model-per-category/create-category-models/script.whizzml

```diff
@@ -27,7 +27,10 @@
 ;; that have the given category.
 (define (category-dataset origin-id field category)
   (let (fltr (flatline "(= (f {{field}}) {{category}})"))
-    (create-dataset origin-id {"lisp_filter" fltr})))
+    (create-dataset origin-id {"lisp_filter" fltr
+                               "name" (str ((fetch origin-id) "name") 
```

Comment:
this function is called for all categories, always with the same dataset, so re-fetching it all the time is  a bit inefficient: we can make this function take a name argument and compute that name only once just before we loop over `category-dataset` below...